golang/cosmos/app/app.go

```diff
@@ -820,15 +827,22 @@ type cosmosInitAction struct {
 // Name returns the name of the App
 func (app *GaiaApp) Name() string { return app.BaseApp.Name() }
 
-func (app *GaiaApp) MustInitController(ctx sdk.Context) {
-	if app.controllerInited {
-		return
+func (app *GaiaApp) CheckControllerInited(expected bool) {
```

Comment:
Insert godoc:

```
// CheckControllerInited exits if the controller initialization state does not match expected.
```

Consider whether you want to exit or panic and do so consistently, here and below.

Reply:
I opted for an Exit here to keep the previous hard exit logic where the `MustInitController` that was run during `BlockingSend` would exit if a failure happened.

I admit I wasn't sure why it exited instead of panic. @michaelfig, any insight?

Reply:
`panic` is recoverable, and indeed, is recovered by the Cosmos SDK in a lot of places.  Rather than having to surgically determine when a panic would or would not be recovered, I just opted for a forced exit when there was an error that should be treated as fatal.

Reply:
@michaelfig I did technically convert some `Exit` into `panic`. More precisely, the `initController` will now `panic` instead of `Exit`. However all `BlockingSend` are guarded by a `CheckControllerInited` that will `Exit` if it wasn't, as a last resort invariant check. I believe that should be sufficient?